blueocean-rest-impl/pom.xml

```diff
@@ -51,9 +51,9 @@
             <groupId>org.jenkins-ci.plugins</groupId>
             <artifactId>ssh-credentials</artifactId>
         </dependency>
-        <dependency>
-            <groupId>com.jcraft</groupId>
-            <artifactId>jsch</artifactId>
+        <dependency> <!-- need this to solve a JSch linkage problem -->
```

Comment:
hey @vivek is this ok? should this be test scoped?

Reply:
this is required, at least for hpi:run. otherwise, it fails with a linkage error due to the jsch libraries between the plugins somehow @michaelneale

Reply:
maybe we can get away with it test scoped? but will it get jsch from the upstream plugin or no? probably @vivek knows for sure :)

Reply:
... tried it, doesn't work.

Reply:
@kzantow Dependence on git-client sounds wrong. can you paste the linkage error you are getting? I am going to try your branch locally anyways to see whats going on.

Reply:
yes this will probably upset things for people downstream... adding this dependency here, so need to find another way.

Reply:
@vivek when the dependency is `jsch`, it gives a linkage error due to both `git-client` and this plugin loading their own copies. You only see it when you try to load a pipeline in the editor using a git pipeline.

Reply:
@kzantow that isn't new functoinality though is it? confused as to why it is picked up now

Reply:
FYI, this is the exception:

```
Caused by: java.lang.LinkageError: loader constraint violation: loader (instance of hudson/ClassicPluginStrategy$AntClassLoader2) previously initiated loading for a different type with name "com/jcraft/jsch/JSch"
	at java.lang.ClassLoader.defineClass1(Native Method)
	at java.lang.ClassLoader.defineClass(ClassLoader.java:763)
	at jenkins.util.AntClassLoader.defineClassFromData(AntClassLoader.java:1139)
	at hudson.ClassicPluginStrategy$AntClassLoader2.defineClassFromData(ClassicPluginStrategy.java:876)
	at jenkins.util.AntClassLoader.getClassFromStream(AntClassLoader.java:1310)
	at jenkins.util.AntClassLoader.findClassInComponents(AntClassLoader.java:1363)
	at jenkins.util.AntClassLoader.findClass(AntClassLoader.java:1326)
	at jenkins.util.AntClassLoader.loadClass(AntClassLoader.java:1079)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	at org.eclipse.jgit.transport.JschConfigSessionFactory.createSession(JschConfigSessionFactory.java:220)
	at org.eclipse.jgit.transport.JschConfigSessionFactory.createSession(JschConfigSessionFactory.java:176)
	at org.eclipse.jgit.transport.JschConfigSessionFactory.getSession(JschConfigSessionFactory.java:110)
	at org.eclipse.jgit.transport.SshTransport.getSession(SshTransport.java:137)
	at org.eclipse.jgit.transport.TransportGitSsh$SshPushConnection.<init>(TransportGitSsh.java:322)
	at org.eclipse.jgit.transport.TransportGitSsh.openPush(TransportGitSsh.java:167)
	at org.eclipse.jgit.transport.PushProcess.execute(PushProcess.java:160)
	at org.eclipse.jgit.transport.Transport.push(Transport.java:1272)
	at org.eclipse.jgit.api.PushCommand.call(PushCommand.java:161)
	at io.jenkins.blueocean.blueocean_git_pipeline.GitUtils.validatePushAccess(GitUtils.java:152)
	at io.jenkins.blueocean.blueocean_git_pipeline.GitBareRepoReadSaveRequest$1.invoke(GitBareRepoReadSaveRequest.java:61)
	at io.jenkins.blueocean.blueocean_git_pipeline.GitBareRepoReadSaveRequest$1.invoke(GitBareRepoReadSaveRequest.java:57)
	at jenkins.plugins.git.GitSCMFileSystem$3.invoke(GitSCMFileSystem.java:193)
	at org.jenkinsci.plugins.gitclient.AbstractGitAPIImpl.withRepository(AbstractGitAPIImpl.java:29)
	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl.withRepository(CliGitAPIImpl.java:71)
	at jenkins.plugins.git.GitSCMFileSystem.invoke(GitSCMFileSystem.java:189)
	at io.jenkins.blueocean.blueocean_git_pipeline.GitCacheCloneReadSaveRequest.invokeOnScm(GitCacheCloneReadSaveRequest.java:149)
	at io.jenkins.blueocean.blueocean_git_pipeline.GitBareRepoReadSaveRequest.read(GitBareRepoReadSaveRequest.java:57)
	at io.jenkins.blueocean.blueocean_git_pipeline.GitReadSaveService.getContent(GitReadSaveService.java:165)
	at io.jenkins.blueocean.rest.impl.pipeline.ScmResourceImpl.getContent(ScmResourceImpl.java:42)
	at java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:627)
	at org.kohsuke.stapler.Function$MethodFunction.invoke(Function.java:343)
	... 103 more

```